lib/fake_names.rb

```diff
@@ -0,0 +1,39 @@
+DISALLOWED_WORDS = [
+  "my", "to", "in", "at", "or", "of", "is", "be", "as", "at", "so", "we", "he", "by", "on", "do", "if", "me", "up",
+  "an", "go", "no", "us", "am", "foo", "foobar", "bar", "and", "one", "two", "three", "four", "five", "six", "seven",
+  "eight", "nine", "ten", "ha", "fake"
+]
+
+module FakeNames
+  def self.is_fake(name)
+    return true if name.nil? or name.empty?
+
+    name.downcase!
+    name.strip!
+
+    allowed_words = []
+
+    words = name.scan(/[a-zA-Z]+/) { |word|
+      allowed_words.push word unless
+        word.include? 'test' or
+        word.include? 'bla' or
+        word.include? 'sdf' or
+        word.bytes.uniq.length == 1 or
+        DISALLOWED_WORDS.include? word
+    }
+
+    allowed_words.empty?
+  end
+
+  def self.is_real(name)
+    !FakeNames.is_fake(name)
```

Comment:
can you use ``self.class.is_fake`` instead?  Feels weird to refer to class name in class.